src/actions/Scroll.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+  SelectionWithEditor,
+} from "../Types";
+import { displaySelectionDecorations } from "../editDisplayUtils";
+import { groupBy } from "../itertools";
+import { commands, TextEditor, Selection, window } from "vscode";
+import { focusEditor } from "./setSelectionsAndFocusEditor";
+
+class Scroll implements Action {
+  targetPreferences: ActionPreferences[] = [{ insideOutsideType: "inside" }];
+
+  constructor(private graph: Graph, private at: string) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    const selectionGroups = groupBy(
+      targets,
+      (t: TypedSelection) => t.selection.editor
+    );
+
+    const selections = <SelectionWithEditor[]>[];
+    const lines = <LineWithEditor[]>[];
+
+    [...selectionGroups].forEach(([editor, targets]) => {
+      const { lineNumber, lineNumberDecorate } = getLineNumber(
+        targets,
+        this.at
+      );
+      lines.push({ lineNumber, editor });
+      selections.push({
+        selection: new Selection(lineNumberDecorate, 0, lineNumberDecorate, 0),
```

Comment:
If you used [`TextLine`](https://code.visualstudio.com/api/references/vscode-api#TextLine)s instead of raw line numbers, you could maybe use `line.range.start` and `line.range.end` to construct the selection.  Up to you tho; not sure it's much of an improvement but just figured I'd point out the API in case you'd not seen it before

Reply:
Thanks but I seen it before. For this use the case since we on really care about one line number not the range I think this worked out well.